.github/workflows/deploy-cargo-docs.yaml

```diff
@@ -0,0 +1,29 @@
+name: Deploy Docs to GitHub Pages
+
+on:
+    push:
+        branches:
+            - main
+
+jobs:
+    deploy-docs:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v2
+
+            - name: Install Rust
+              uses: actions-rs/toolchain@v1
+              with:
+                  toolchain: stable
+                  profile: minimal
+                  override: true
+
+            - name: Generate Documentation
+              run: cargo doc --no-deps
```

Comment:
Can you run `make docs` instead? At this moment it is completely equivalent but in the future we might add things or options to the makefile